ddtrace/tracer/textmap.go

```diff
@@ -196,6 +202,9 @@ func getPropagators(cfg *PropagatorConfig, env string) []Propagator {
 // out of the current process. The implementation propagates the
 // TraceID and the current active SpanID, as well as the Span baggage.
 func (p *chainedPropagator) Inject(spanCtx ddtrace.SpanContext, carrier interface{}) error {
+	if c, ok := carrier.(*SQLCommentCarrier); ok {
```

Comment:
I think you've actually solved this "hack": we don't need this change because one can simply define an `SQLCarrier` and call `Inject` on it since you've made it public.